lib/sustentation/issues/issues.ex

```diff
@@ -0,0 +1,104 @@
+defmodule Sustentation.Issues do
+  @moduledoc """
+  The Issues context.
+  """
+
+  import Ecto.Query, warn: false
+  alias Sustentation.Repo
+
+  alias Sustentation.Issues.Category
+
+  @doc """
+  Returns the list of categories.
+
+  ## Examples
+
+      iex> list_categories()
+      [%Category{}, ...]
+
+  """
+  def list_categories do
+    Repo.all(Category)
+  end
+
+  @doc """
+  Gets a single category.
+
+  Raises `Ecto.NoResultsError` if the Category does not exist.
+
+  ## Examples
+
+      iex> get_category!(123)
+      %Category{}
+
+      iex> get_category!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_category!(id), do: Repo.get!(Category, id)
```

Comment:
Uma alteração que eu acho interessante é mudar `get_category!` para `get_category` e fazer ele retornar `{:ok, category}` ou `{:error, :not_found}` ao invés de subir uma exception. O motivo é que uma função que retorna `:ok | :not_found` comunica que é uma função que pode falhar e fica mais fácil de fazer pattern matching e de testar.

Essa função `Repo.get!` levanta uma exception `Ecto.NoResultError`, o Phoenix quando pega uma exceção dessa vai tratar ela e fazer retornar uma view de error do fallback controller, como se fosse um `{:error, :not_found}`.

Não esquece de alterar a doc acima!

```suggestion
  def get_category(id) do
    case Repo.get(Category, id) do
      %Category{} = category -> {:ok, category}
      _ -> {:error, :not_found}
    end
```